src/main/java/com/godaddy/sonar/ruby/simplecovrcov/SimpleCovRcovJsonParserImpl.java

```diff
@@ -11,49 +16,78 @@
 
 import java.io.File;
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
 import java.util.Map;
 
 public class SimpleCovRcovJsonParserImpl implements SimpleCovRcovJsonParser {
     private static final Logger LOG = Loggers.get(SimpleCovRcovJsonParserImpl.class);
 
     public Map<String, CoverageMeasuresBuilder> parse(File file) throws IOException {
-        Map<String, CoverageMeasuresBuilder> coveredFiles = Maps.newHashMap();
+        CoverageReport coverageReport = readAndParseReportFile(file);
+        return processCoverageReport(coverageReport);
+    }
 
-        File fileToFindCoverage = file;
+    private CoverageReport readAndParseReportFile(File file) throws IOException {
+        CoverageReport coverageReport = new CoverageReport();
 
-        String fileString = FileUtils.readFileToString(fileToFindCoverage, "UTF-8");
+        String fileString = FileUtils.readFileToString(file, "UTF-8");
 
         JsonParser parser = new JsonParser();
         JsonObject resultJsonObject = parser.parse(fileString).getAsJsonObject();
-        JsonObject coverageJsonObj = resultJsonObject.get("MiniTest").getAsJsonObject().get("coverage").getAsJsonObject();
 
-        // for each file in the coverage report
-        for (int j = 0; j < coverageJsonObj.entrySet().size(); j++) {
-            CoverageMeasuresBuilder fileCoverage = CoverageMeasuresBuilder.create();
+        for(Map.Entry coverageMapEntry : resultJsonObject.entrySet()){
+            coverageReport.addReporter(buildReporter(coverageMapEntry));
+        }
+        return coverageReport;
+    }
 
-            String filePath = ((Map.Entry) coverageJsonObj.entrySet().toArray()[j]).getKey().toString();
-            LOG.debug("filePath " + filePath);
+    private Map<String, CoverageMeasuresBuilder> processCoverageReport(CoverageReport coverageReport) {
+        if (coverageReport.hasSeveralReporters()) {
+            LOG.warn("Coverage report has several reporters! Notice, please, that only the first one will be considered by Sonar!");
```

Comment:
Mostly it was a temporary solution from my side to not rely only on MiniTest or RSpec reports. I am not sure how SonarQube tackle with several reports. But maybe it is okay, because some part of code could be covered by MiniTest unit tests and another part by RSpec. I think that it is correct to add coverage marks from all reporters, right?